packages/jest-jasmine2/src/jasmine/Spec.js

```diff
@@ -121,10 +127,26 @@ Spec.prototype.onException = function onException(e) {
     return;
   }
 
+  const {expected, actual} = e;
+  let message;
+  if (expected && actual) {
+    const diffString = diff(expected, actual, {
+      expand: this.expand,
+    });
+    message = matcherHint('.toBe') +
+      '\n\n' +
+      `Expected value to be (using ===):\n` +
+      `  ${printExpected(expected)}\n` +
+      `Received:\n` +
+      `  ${printReceived(actual)}` +
+      (diffString ? `\n\nDifference:\n\n${diffString}` : '');
+  }
```

Comment:
Any way we can make the call to `diff` and the message lazy, as in `message = () => {…}`? Otherwise I'm a bit worried here about perf in the case where we don't actually need the message.